src/YQComboBox.cc

```diff
@@ -122,11 +121,21 @@ void YQComboBox::addItem( YItem * item )
 
     if ( item->hasIconName() )
     {
-	string iconName = iconFullPath( item );
-	icon = QIcon( iconName.c_str() );
+        if (QIcon::hasThemeIcon(item->iconName().c_str()))
```

Comment:
This block is copied verbatim 11 times (!). This needs to be factored out.

Please make this a public function in YQUI so it is usable from everywhere:

```C++

QIcon YQUi::loadIcon( const QString & iconName )
{
    QIcon icon;

    if ( QIcon::hasThemeIcon( iconName ) )
    ...

    return icon;
}
```

Also please note our C++ coding style: One blank after each opening `(`, one before each closing `)`, so it's

```C++
    doSomething( doMore( arg1, arg2, arg3 ) );
```

Reply:
Yeah, you are quite right, I will do that in a few hours.

Reply:
Forgot to add `const` to the function (since it doesn't change the YQUI object):

```C++
QIcon YQUi::loadIcon( const QString & iconName ) const
```